File: 9-9.py
def is_prime(x):
    if x <= 1:
        return False
    for i in range(2, x + 1):
        if i * i > x:
            return True
        if x % i == 0:
            return False

def solution(n, k):
    answer = 0

    P = ""
    while n > 0:
        d = n % k
        n = n // k
        P += str(d)
    P = P[ : : -1]

    for x in P.split('0'):
        if x != '' and is_prime(int(x)):
            answer += 1

    return answer

File: 9-9.cpp
#include <bits/stdc++.h>
using namespace std;

bool is_prime(long long x) {
    if (x <= 1) return false;
    for (long long i = 2; i * i <= x; i++)
        if (x % i == 0) return false;
    return true;
}

int solution(int n, int k) {
    int answer = 0;
    while (n > 0) {
        long long P = 0, x = 1;
        while (n > 0) {
            int d = n % k;
            n = n / k;
            if (d == 0) break;
            P += d * x;
            x *= 10;
        }
        if (is_prime(P))
            answer++;
    }
    return answer;
}